transport/conn.go

```diff
@@ -376,6 +376,10 @@ func OpenShardConn(ctx context.Context, addr string, si ShardInfo, cfg ConnConfi
 	it := ShardPortIterator(si)
 	maxTries := (maxPort-minPort+1)/int(si.NrShards) + 1
 	for i := 0; i < maxTries; i++ {
+		if err := ctx.Err(); err != nil {
+			return nil, fmt.Errorf("aborted opening connection on shard %d: %w", si.Shard, err)
+		}
+
```

Comment:
The fix this commit introduces doesn't seem to be related to the main purpose of the PR (adding integration tests to the CI). It should be sent in a separate PR.

Reply:
Will move that to #271 as its related to logs, dropped